dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/tools/ParallelDependentTestFinder.java

```diff
@@ -0,0 +1,541 @@
+
+        if (!tests.isEmpty()) {
+            if (isOriginalOrder) {
+                addDependency(tests.get(0), revealed, revealingOrder, true);
+            } else {
+                addDependency(tests.get(0), revealed, revealingOrder, false);
+            }
+        }
+    }
+
+    /**
+     * @return A set of tests that must come AFTER the test in question.
+     */
+    private Set<String> getAfterDependencies(final String testName) {
+        return knownDependencies.entrySet().stream()
+                .filter(dependencies -> dependencies.getValue().containsAfter(testName))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+    }
+
+    public static String dependenciesToString(final List<ParallelDependentTestFinder> dependencies) {
+        return dependencies.stream()
+                .map(ParallelDependentTestFinder::toString)
+                .reduce("", String::concat);
+    }
+
+    @Override
+    public String toString() {
+        final TestData testData = knownDependencies.get(dependentTestName);
+
+        if (testData != null) {
+            return testData.toString();
+        } else {
+            return "";
+        }
+    }
+
+    private String debugLogToString() {
+        return debugLog.stream()
+                .map(TestData::toString)
+                .collect(Collectors.joining("\n"));
+    }
+
+    public static void writeToFile(final ParallelDependentTestFinder dtFinder,
+                                   final boolean includeDebugLog) {
+        writeToFile(Collections.singletonList(dtFinder), includeDebugLog);
+    }
+
+    public static void writeToFile(final ParallelDependentTestFinder dtFinder,
+                                   final File outputFile,
+                                   final boolean includeDebugLog) {
+        writeToFile(Collections.singletonList(dtFinder), outputFile, includeDebugLog);
+    }
+
+    public static void writeToFile(final List<ParallelDependentTestFinder> dtFinders,
+                                   final boolean includeDebugLog) {
+        writeToFile(dtFinders, DT_FILE, includeDebugLog);
+    }
+
+    /**
+     * Writes all the dependency information to a file.
+     * @param dtFinders The dtFinders that contain the dependency information we need.
+     */
+    public static void writeToFile(final List<ParallelDependentTestFinder> dtFinders,
+                                   final File outputFile,
+                                   final boolean includeDebugLog) {
+        if (outputFile != null) {
+            FileWriter output = null;
+            BufferedWriter writer = null;
+            try {
+                output = new FileWriter(outputFile);
+                writer = new BufferedWriter(output);
+
+                for (final ParallelDependentTestFinder dtFinder : dtFinders) {
+                    if (includeDebugLog) {
+                        writer.write(dtFinder.debugLogToString() + "\n");
+                    }
```

Comment:
The debug log and the actual printing of the precomputed dependencies should be separate. The precomputed dependencies only cares about what the dependent test is and what tests it depends on to come before and after it. The debug log should have all fields of the TestData objects be printed. 

Its fine to have a method that does both but the debug log does not need the info of the precomputed dependencies (it should be a else here) and this method should reside in the Runner class and take a set of TestData objects.